### PR TITLE
Fix #1947, #1873: FabricBlockSettings#copy does not copy all settings

### LINKED ISSUE #1947: FabricBlockSettings generic constructor is missing copy methods for certain settings
The FabricBlockSettings constructor does not copy values for the following attributes:
- `allowsSpawningPredicate`
- `solidBlockPredicate`
- `suffocationPredicate`
- `blockVisionPredicate`
- `postProcessPredicate`
- `emissiveLightingPredicate`

This would also require new mixin getters for the `AbstractBlockSettingsAccessor`

Issue on the following version:
```properties
yarn_mappings=1.18.1+build.14
fabric_version=0.45.0+1.18
```

### PR DESCRIPTION
**Pull request by Luligabi1**

### CODE REVIEW

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/mixin/object/builder/AbstractBlockSettingsAccessor.java [hunk 1]
```diff
@@ -19,6 +19,8 @@
 import java.util.function.Function;
 import java.util.function.ToIntFunction;
 
+import net.minecraft.entity.EntityType;
+
```
Comment (modmuss50) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/mixin/object/builder/AbstractBlockSettingsAccessor.java]: Please ensure this follows the checkstyle rules.

**Review by deirn — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Comment by modmuss50** [comment]
This is failing to compile.

**Comment by Luligabi1** [comment]
> This is failing to compile.

Looks like some things broke when upgrading to 1.18.2. It _should_ be fine now.

**Comment by Player3324** [comment]
```
 > Task :fabric-object-builder-api-v1:compileJava
/__w/fabric/fabric/fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricBlockSettings.java:76: error: cannot find symbol
		BlockSettingsInternals otherInternals = (BlockSettingsInternals) settings;
		^
  symbol:   class BlockSettingsInternals
  location: class FabricBlockSettings
/__w/fabric/fabric/fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricBlockSettings.java:76: error: cannot find symbol
		BlockSettingsInternals otherInternals = (BlockSettingsInternals) settings;
		                                         ^
  symbol:   class BlockSettingsInternals
  location: class FabricBlockSettings
/__w/fabric/fabric/fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricBlockSettings.java:77: error: package FabricBlockInternals does not exist
		FabricBlockInternals.ExtraData extraData = otherInternals.getExtraData();
		                    ^
/__w/fabric/fabric/fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricBlockSettings.java:80: error: cannot find symbol
			((BlockSettingsInternals) this).setExtraData(extraData);
			  ^
  symbol:   class BlockSettingsInternals
  location: class FabricBlockSettings
  ```

**Pull request closed without merging**